src/main/java/org/folio/fql/service/FqlValidationService.java

```diff
@@ -34,41 +44,67 @@ public Map<String, String> validateFql(
     }
   }
 
-  private Map<String, String> assertFqlFieldPresentInEntityType(
-    FqlCondition<?> fqlCondition,
-    EntityType entityType
-  ) {
+  private Map<String, String> assertFqlFieldPresentInEntityType(FqlCondition<?> fqlCondition, EntityType entityType) {
     Map<String, String> errorMap = new HashMap<>();
     if (fqlCondition instanceof AndCondition andCondition) {
       andCondition
         .value()
-        .forEach(subCondition ->
-          errorMap.putAll(
-            assertFqlFieldPresentInEntityType(subCondition, entityType)
-          )
-        );
+        .forEach(subCondition -> errorMap.putAll(assertFqlFieldPresentInEntityType(subCondition, entityType)));
     } else {
+      FqlField field = ((FieldCondition<?>) fqlCondition).field();
+
       entityType
         .getColumns()
         .stream()
-        .filter(col ->
-          col.getName().equals(((FieldCondition<?>) fqlCondition).fieldName())
-        )
+        .filter(col -> checkColumnMatch(field, col))
         .findFirst()
         .ifPresentOrElse(
           value -> {},
           () -> {
-            String fieldName = ((FieldCondition<?>) fqlCondition).fieldName();
+            String fieldName = field.serialize();
             errorMap.put(
               fieldName,
-              "Field " +
-              fieldName +
-              " is not present in definition of entity type " +
-              entityType.getName()
+              "Field " + fieldName + " is not present in definition of entity type " + entityType.getName()
             );
           }
         );
     }
     return errorMap;
   }
+
+  public static boolean checkColumnMatch(FqlField field, EntityTypeColumn col) {
+    if (!col.getName().equals(field.getColumnName())) {
+      return false;
+    }
+
+    Deque<SubField> subFields = new ArrayDeque<>(field.getSubFields());
+    EntityDataType dataType = col.getDataType();
+
+    while (!subFields.isEmpty()) {
+      SubField subField = subFields.pop();
+      if (subField instanceof ArraySubField) {
+        if (dataType instanceof ArrayType arrayDataType) {
+          dataType = arrayDataType.getItemDataType();
+        } else {
+          return false;
+        }
+      } else if (subField instanceof PropertySubField propertySubField) {
```

Comment:
Technically this `else if` is never false, but I feel like having this makes it more readable (and will also help if this logic is ever extended in the future).